pyramid_swagger/api.py

```diff
@@ -4,66 +4,85 @@
 """
 import simplejson
 
+from pyramid_swagger.model import PyramidEndpoint
 
-def register_api_doc_endpoints(config):
-    """Create and register pyramid endpoints for /api-docs*."""
-    swagger_schema = config.registry.settings['pyramid_swagger.schema']
-    register_resource_listing(config, swagger_schema.resource_listing)
 
-    for name, filepath in swagger_schema.api_declarations.items():
-        with open(filepath) as input_file:
-            register_api_declaration(
-                config,
-                name,
-                simplejson.load(input_file)
-            )
+# TODO: document that this is now a public interface
+def register_api_doc_endpoints(config, endpoints, base_path='/api-docs'):
+    """Create and register pyramid endpoints to service swagger api docs.
+    Routes and views will be registered on the `config` at `path`.
+
+    :param config: a pyramid configuration to register the new views and routes
+    :type  config: :class:`pyramid.config.Configurator`
+    :param endpoints: a list of endpoints to register as routes and views
+    :type  endpoints: a list of :class:`pyramid_swagger.model.PyramidEndpoint`
+    :param base_path: the base path used to register api doc endpoints.
+        Defaults to `/api-docs`.
+    :type  base_path: string
+    """
+    for endpoint in endpoints:
+        path = base_path + endpoint.path
```

Comment:
It would be a good idea to `basepath.rstrip('/')` when allowing user to enter custom basePath